packages/forms-web-app/src/dynamic-forms/validator/validator.js

```diff
@@ -1,49 +1,34 @@
-// const { body, validationResult } = require('express-validator');
-// const hasQuestionnaire = require('../definitions/questionnaire/hasQuestionnaire');
-// const questionnaire = hasQuestionnaire.questionnaire;
-// const formHelper = require('../lib/dynamicformhelper');
+const RequiredValidator = require('./requiredValidator');
+const ValidOptionValidator = require('./validOptionValidator');
+const hasJourney = require('../has-questionnaire/journey');
 
-// const validate = () => {
-//     return async (req, res, next) => {
-//         const { appealId, section, question } = req.params;
-//         var questionObj = formHelper.getQuestionBySectionAndName(questionnaire, section, question);
-//         const validations = [];
-//         switch (questionObj.validator?.type) {
-//             case "text":
-//                 validations.push(textValidations(questionObj))
-//                 break;
-//             case "boolean":
-//                 validations.push(booleanValidations(questionObj))
-//                 break;
-//             case "boolean-text":
-//                 validations.push(booleanValidations(questionObj))
-//                 validations.push(textValidations(questionObj))
-//                 break;
-//         }
-//         await Promise.all(validations.map(validation => validation.run(req)));
+const validate = () => {
+	return async (req, res, next) => {
+		const { section, question } = req.params;
+		var questionObj = hasJourney.getQuestionBySectionAndName(section, question);
+		const validations = [];
+		questionObj.validators.forEach((validator) => {
+			switch (validator.constructor) {
+				case RequiredValidator:
+					validations.push(new RequiredValidator().validate(questionObj));
+					break;
+				case ValidOptionValidator:
+					validations.push(new ValidOptionValidator().validate(questionObj));
+					break;
+				default:
+					throw new Error(`Validator of type ${validator.constructor} not implemented`);
+			}
+		});
 
-//        next();
-//     };
-// };
+		const result = await Promise.all(validations.map((validation) => validation.run(req)));
```

Comment:
We'll need to do this in some kind of loop instead https://github.com/Planning-Inspectorate/appeal-planning-decision/blob/spike/dynamic-questionnaire-renderer-validation-examples/packages/questionnaire-web-app/src/validator/validator.js#L12 so we can check if there are any errors after each validation and exit out the validation middleware once any errors have been raised, just so we can support controlling the order of validations and avoid running any validation rules that don't need to be evaluated because an earlier validation has failed.